src/imu_control_parameters.cpp

```diff
@@ -0,0 +1,324 @@
+    m_funcMapGet["accel_calibbias_y"] = &IIOWrapper::accel_y_calibbias;
+    m_funcMapGet["accel_calibbias_z"] = &IIOWrapper::accel_z_calibbias;
+    m_funcMapGet["anglvel_calibbias_x"] = &IIOWrapper::anglvel_x_calibbias;
+    m_funcMapGet["anglvel_calibbias_y"] = &IIOWrapper::anglvel_y_calibbias;
+    m_funcMapGet["anglvel_calibbias_z"] = &IIOWrapper::anglvel_z_calibbias;
+    m_funcMapGet["sampling_frequency"] = &IIOWrapper::sampling_frequency;
+}
+
+bool ImuControlParameters::setParameters(std::map<std::string, int32_t>& valueMap)
+{
+    bool hasError = false;
+
+    std::vector<std::string> keys;
+    for (auto it = m_funcMap.begin(); it != m_funcMap.end(); it++)
+        keys.push_back(it->first);
+
+    for (const std::string &key: keys)
+    {
+        int deviceParamValue = (m_iioWrapper.*(m_funcMapGet[key]))();
+
+        if(deviceParamValue != valueMap[key])
+        {
+            const char * ckey = key.c_str();
+
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_parameter_before_set"), "%s: '%d'", ckey , deviceParamValue);
+
+            int ret = (m_iioWrapper.*(m_funcMap[key]))(valueMap[key]);
+
+            if(ret == 0)
+            {
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "%s: '%d'", ckey , valueMap[key]);
+            }
+            else
+            {
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter_error"), "error on set parameter %s: '%d' ", ckey , valueMap[key]);
+                hasError = true;
+            }
+        }
+    }
+    return hasError;
+}
+
+bool ImuControlParameters::triggerCommand(std::string& command_to_execute, bool hasError)
+{
+    std::vector<std::string> keys;
+    for (auto it = m_funcMap2.begin(); it != m_funcMap2.end(); it++)
+        keys.push_back(it->first);
+
+    for (const std::string &key: keys)
+    {
+        if(key == command_to_execute)
+        {
+            const char * ckey = key.c_str();
+
+            int ret = (m_iioWrapper.*(m_funcMap2[key]))();
+
+            if(ret == 0)
+            {
+                m_node->set_parameter(rclcpp::Parameter("command_to_execute", "no_command"));
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "command_to_execute: '%s'", ckey);
+                break;
+            }
+            else
+            {
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter_error"), "command_to_execute error on set %s", ckey );
+                hasError = true;
+            }
+        }
+    }
+    return hasError;
+}
+
+void ImuControlParameters::userParamSettingMode(std::map<std::string, int32_t>& valueMap,
+                                                int& lastOperationMode,
+                                                bool& wasConfMode)
+{
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp_configuration_mode"), "Set ros parameters");
+
+    int32_t filter_size = m_node->get_parameter("filter_size").get_parameter_value().get<int32_t>();
```

Comment:
Can we store this variables directly into the map without the extra parameters declaration?

Reply:
Or even better, can this be replaced with a for loop to get each parameter from the list? Or are there some parameters in the list which are not required here?